src/sources/postgresql_metrics.rs

```diff
@@ -61,6 +61,10 @@ macro_rules! gauge {
 
 #[derive(Debug, Snafu)]
 enum BuildError {
+    #[snafu(display(
+        "null database can not be specified in both `include_databases` and `exclude_databases`"
```

Comment:
Can you give me an example of what the config would look like that would trigger this error? Also it looks like `datname` is not nullable in the Postgres schema (at least in version 13) so maybe we should explicitly reject `null`s?

Reply:
Example:
```
include_databases: [""]
exclude_databases: [""]
```
Postgres still have nullable `datname`:
```
# SELECT COUNT(1) FROM pg_stat_database WHERE datname IS NULL;
┌───────┐
│ count │
├───────┤
│     1 │
└───────┘
(1 row)
```

Reply:
Aha! Interesting. I ran `\d+` in psql and it didn't seem to be nullable, but I do see it returns a record.

Seems to be for tracking objects not in a database: https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=77bd49adba4711b4497e7e39a5ec3a9812cbd52a

Regarding the example, we've typically had `exclude_*` options take precedence over `include_*` options so that, for example, if you did:

```
include_databases = ["foobar"]
exclude_databases = ["foobar"]
```

You'd effectively get no metrics. I think we can apply the same logic here and just have `exclude` override rather than returning an error in this case.

Reply:
Changed.